File: RAI/Clase 14/Java/Moneda.java
package Java;

public class Moneda implements Divisa{
    private String nombre;
    private String pais;
    private double ref;

    public Moneda(String nombre, String pais, double ref){
        this.nombre = nombre;
        this.pais = pais;
        this.ref = ref;
    }

    public String getNombre(){
        return this.nombre;
    }

    public String getPais(){
        return this.pais;
    }

    public double getRef(){
        return this.ref;
    }

    @Override
    public double cambiarPor(double monto, Divisa divisa){
        double aDolar = ((Moneda) divisa).getRef() * monto; //((Moneda) divisa) cambia el tipo solamente para la operación de Divisa por Moneda
        double cambio = this.getRef() / aDolar;
        return cambio;
    }
}

File: RAI/Clase 14/Python/Moneda.py
from Divisa import Divisa

class Moneda(Divisa):
    
    def __init__(self, nombre, pais, ref):
        self.nombre = nombre
        self.pais = pais
        self.ref = ref
        
    def get_nombre(self):
        return self.nombre
    
    def get_pais(self):
        return self.pais
    
    def get_ref(self):
        return self.ref
    
    def cambiar_por(self, monto, divisa):
        a_dolar = divisa.get_ref() * monto
        cambio = self.get_ref()/ a_dolar
        return cambio
        